### DISCUSSION TITLE
rules.unique V5

### DISCUSSION THREAD

**Discussion by sidicastro** [adonisjs/core#2579] [Help]
when trying to update a form where there is a field with rules.unique the validator complains.
PS, I want to update other fields on the form without changing the field with rules.unique.

there is some way to:
so check the rules.unique (in the update only), if it has been changed?

**Comment by RomainLanz** [comment]
Hey @sidicastro! 👋 

You should use an additional constraints.
➡️ https://docs.adonisjs.com/reference/validator/rules/unique#additional-constraints

**Reply by sidicastro** [reply]
Yes, using additional restrictions like
rules.unique({ table: 'products', column: 'code', where: { id: this.ctx.params} })

solves the problem in the update but then it breaks if it is in the INSERT.
I use a rules.unique validator for both situations: Insertion and Update, when inpinho restrictions solves the problem if it has updating but breaks if it is in the Insertion.

**Comment by victorLessa** [comment]
I have the same problem

**Comment by thetutlage** [comment]
The validators should be different for update and insert.

**Reply by sidicastro** [reply]
THAT'S RIGHT! That's what I did: A validator for each verb

**Reply by thetutlage** [reply]
Can you please share the complete code for the validator, along with the expected outcome and the actual outcome

**Comment by arthurcvm** [comment]
I need refer another field value in custom rule, but I do not know get the reference to validate.
